cpp/bench/ann/src/hnswlib/hnswlib_wrapper.h

```diff
@@ -159,10 +160,12 @@ void HnswLib<T>::build(const T* dataset, size_t nrow, cudaStream_t)
 template <typename T>
 void HnswLib<T>::set_search_param(const AnnSearchParam& param_)
 {
-  auto param     = dynamic_cast<const SearchParam&>(param_);
-  appr_alg_->ef_ = param.ef;
+  auto param        = dynamic_cast<const SearchParam&>(param_);
+  appr_alg_->ef_    = param.ef;
+  metric_objective_ = param.metric_objective;
 
-  if (!thread_pool_ || num_threads_ != param.num_threads) {
+  if (metric_objective_ != Objective::LATENCY &&
+      (!thread_pool_ || num_threads_ != param.num_threads)) {
```

Comment:
if `param.num_threads == 1` - is it worth it to create the threadpool here?

Reply:
Good point- it's simple enough to test for that here.